tkg/client/upgrade_cluster.go

```diff
@@ -1145,6 +1146,13 @@ func (c *TkgClient) PatchKubernetesVersionToKubeadmControlPlane(regionalClusterC
 		}
 	}
 
+	if semver.Compare(clusterUpgradeConfig.UpgradeComponentInfo.KubernetesVersion, "v1.24.0") >= 0 {
```

Comment:
I hope this integration to enable it for v1.24 is as intended in tanzu? Happy to refactor if this is the wrong place to implement the functionality.